packages/core/src/messages/responses/storage/scalar/storage-get.ts

```diff
@@ -0,0 +1,167 @@
+import {StorageItemType, StorageGetResponse} from '../../enums';
+import {BaseResponseError, ResponseBase} from '../../response-base';
+import {SdkError} from '../../../../errors';
+
+interface IResponse {
+  readonly type: StorageGetResponse;
+  value(): string | number | Uint8Array | undefined;
+}
+
+export abstract class Success extends ResponseBase implements IResponse {
+  readonly type: StorageGetResponse.Success;
+  readonly itemType: StorageItemType;
+
+  abstract value(): string | number | Uint8Array;
+
+  abstract valueInt(): number | undefined;
+
+  abstract valueDouble(): number | undefined;
+
+  abstract valueString(): string | undefined;
+
+  abstract valueBytes(): Uint8Array | undefined;
```

Comment:
I want to see what you all and @malandis end up thinking about what I did on the Java PR, where I pulled these value types out into their own class, so that at the `Success` level there was only a single `Value`. If we do that in Java we probably want to do it here.